Cloud Client Builder breaks if any of the normal builder methods is used

**Describe the bug**

I use the cloud client builder to configure my cluster id, credentials and then want so set some new default values, i.e. job timeout. Then the client build is not configuring the credentials provider and gateway address correctly, therefore I cannot connect to my cluster

**To Reproduce**

```
ZeebeClient.newCloudClientBuilder()
            .withClusterId("862e696b-0f3e-4452-a9ad-83b825d7f33c")
            .withClientId("9mXzhnpjmteNKJ0PvPsH23IATDBl1E2r")
            .withClientSecret("vD4oQ2kBH4pvw_ooSwn1IFW_qy7lb0f.WFtoP9YIboFZ1S2bRakjvobUN8211Lbm")
            .defaultJobTimeout(Duration.ofSeconds(100))
            .build()
```

**Expected behavior**

I can access my cluster with custom default configurations.

**Log/Stacktrace**


<details><summary>Full Stacktrace</summary>
 <p>

```
io.camunda.zeebe.client.api.command.ClientStatusException: io exception

	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:93)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:50)
	at io.camunda.zeebe.client.AltClusterTest.test(AltClusterTest.java:17)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:221)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:48)
	... 26 more
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:478)
	at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:463)
	at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:427)
	at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:460)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:553)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:68)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:739)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:718)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /0.0.0.0:26500
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:707)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

</p>
</details>

**Root Cause**

The issue is that the `ZeebeClientCloudBuilderImpl` returns the inner wrapped builder, instead of itself, therefore we lose the configuration of the client cloud builder.

https://github.com/camunda-cloud/zeebe/blob/eda7a263077e612daa9c48f9843591d2826f5486/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java#L113-L116


**Environment:**
 - OS: Linux
 - Zeebe Version: 1.0.0
 - Configuration: Java Client

--- a/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java
+++ b/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java
@@ -95,6 +95,7 @@ public class ZeebeClientCloudBuilderImpl
     if (properties.containsKey(ClientProperties.CLOUD_CLIENT_SECRET)) {
       withClientSecret(properties.getProperty(ClientProperties.CLOUD_CLIENT_SECRET));
     }
+    innerBuilder.withProperties(properties);
     return this;
   }
 

--- a/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java
+++ b/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java
@@ -21,7 +21,6 @@ import static io.camunda.zeebe.client.impl.command.ArgumentUtil.ensureNotNull;
 import io.camunda.zeebe.client.ClientProperties;
 import io.camunda.zeebe.client.CredentialsProvider;
 import io.camunda.zeebe.client.ZeebeClient;
-import io.camunda.zeebe.client.ZeebeClientBuilder;
 import io.camunda.zeebe.client.ZeebeClientCloudBuilderStep1;
 import io.camunda.zeebe.client.ZeebeClientCloudBuilderStep1.ZeebeClientCloudBuilderStep2;
 import io.camunda.zeebe.client.ZeebeClientCloudBuilderStep1.ZeebeClientCloudBuilderStep2.ZeebeClientCloudBuilderStep3;
@@ -67,22 +66,26 @@ public class ZeebeClientCloudBuilderImpl
   }
 
   @Override
-  public ZeebeClientBuilder gatewayAddress(final String gatewayAddress) {
-    return innerBuilder.gatewayAddress(gatewayAddress);
+  public ZeebeClientCloudBuilderStep4 gatewayAddress(final String gatewayAddress) {
+    innerBuilder.gatewayAddress(gatewayAddress);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder usePlaintext() {
-    return innerBuilder.usePlaintext();
+  public ZeebeClientCloudBuilderStep4 usePlaintext() {
+    innerBuilder.usePlaintext();
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder credentialsProvider(final CredentialsProvider credentialsProvider) {
-    return innerBuilder.credentialsProvider(credentialsProvider);
+  public ZeebeClientCloudBuilderStep4 credentialsProvider(
+      final CredentialsProvider credentialsProvider) {
+    innerBuilder.credentialsProvider(credentialsProvider);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder withProperties(final Properties properties) {
+  public ZeebeClientCloudBuilderStep4 withProperties(final Properties properties) {
     if (properties.containsKey(ClientProperties.CLOUD_CLUSTER_ID)) {
       withClusterId(properties.getProperty(ClientProperties.CLOUD_CLUSTER_ID));
     }
@@ -92,62 +95,73 @@ public class ZeebeClientCloudBuilderImpl
     if (properties.containsKey(ClientProperties.CLOUD_CLIENT_SECRET)) {
       withClientSecret(properties.getProperty(ClientProperties.CLOUD_CLIENT_SECRET));
     }
-    return innerBuilder.withProperties(properties);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder defaultJobWorkerMaxJobsActive(final int maxJobsActive) {
-    return innerBuilder.defaultJobWorkerMaxJobsActive(maxJobsActive);
+  public ZeebeClientCloudBuilderStep4 defaultJobWorkerMaxJobsActive(final int maxJobsActive) {
+    innerBuilder.defaultJobWorkerMaxJobsActive(maxJobsActive);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder numJobWorkerExecutionThreads(final int numThreads) {
-    return innerBuilder.numJobWorkerExecutionThreads(numThreads);
+  public ZeebeClientCloudBuilderStep4 numJobWorkerExecutionThreads(final int numThreads) {
+    innerBuilder.numJobWorkerExecutionThreads(numThreads);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder defaultJobWorkerName(final String workerName) {
-    return innerBuilder.defaultJobWorkerName(workerName);
+  public ZeebeClientCloudBuilderStep4 defaultJobWorkerName(final String workerName) {
+    innerBuilder.defaultJobWorkerName(workerName);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder defaultJobTimeout(final Duration timeout) {
-    return innerBuilder.defaultJobTimeout(timeout);
+  public ZeebeClientCloudBuilderStep4 defaultJobTimeout(final Duration timeout) {
+    innerBuilder.defaultJobTimeout(timeout);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder defaultJobPollInterval(final Duration pollInterval) {
-    return innerBuilder.defaultJobPollInterval(pollInterval);
+  public ZeebeClientCloudBuilderStep4 defaultJobPollInterval(final Duration pollInterval) {
+    innerBuilder.defaultJobPollInterval(pollInterval);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder defaultMessageTimeToLive(final Duration timeToLive) {
-    return innerBuilder.defaultMessageTimeToLive(timeToLive);
+  public ZeebeClientCloudBuilderStep4 defaultMessageTimeToLive(final Duration timeToLive) {
+    innerBuilder.defaultMessageTimeToLive(timeToLive);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder defaultRequestTimeout(final Duration requestTimeout) {
-    return innerBuilder.defaultRequestTimeout(requestTimeout);
+  public ZeebeClientCloudBuilderStep4 defaultRequestTimeout(final Duration requestTimeout) {
+    innerBuilder.defaultRequestTimeout(requestTimeout);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder caCertificatePath(final String certificatePath) {
-    return innerBuilder.caCertificatePath(certificatePath);
+  public ZeebeClientCloudBuilderStep4 caCertificatePath(final String certificatePath) {
+    innerBuilder.caCertificatePath(certificatePath);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder keepAlive(final Duration keepAlive) {
-    return innerBuilder.keepAlive(keepAlive);
+  public ZeebeClientCloudBuilderStep4 keepAlive(final Duration keepAlive) {
+    innerBuilder.keepAlive(keepAlive);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder withInterceptors(final ClientInterceptor... interceptor) {
-    return innerBuilder.withInterceptors(interceptor);
+  public ZeebeClientCloudBuilderStep4 withInterceptors(final ClientInterceptor... interceptor) {
+    innerBuilder.withInterceptors(interceptor);
+    return this;
   }
 
   @Override
-  public ZeebeClientBuilder withJsonMapper(final JsonMapper jsonMapper) {
-    return innerBuilder.withJsonMapper(jsonMapper);
+  public ZeebeClientCloudBuilderStep4 withJsonMapper(final JsonMapper jsonMapper) {
+    innerBuilder.withJsonMapper(jsonMapper);
+    return this;
   }
 
   @Override
@@ -177,6 +191,10 @@ public class ZeebeClientCloudBuilderImpl
       ensureNotNull("client id", clientId);
       ensureNotNull("client secret", clientSecret);
       final OAuthCredentialsProviderBuilder builder = new OAuthCredentialsProviderBuilder();
+
+      if (innerBuilder.isPlaintextConnectionEnabled()) {
+        Loggers.LOGGER.debug("Expected setting 'usePlaintext' to be 'false', but found 'true'.");
+      }
       return builder
           .audience(String.format("%s.%s", clusterId, BASE_ADDRESS))
           .clientId(clientId)

